src/Datastore/Entity.php

```diff
@@ -0,0 +1,224 @@
+<?php
+/**
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Cloud\Datastore;
+
+use ArrayAccess;
+use InvalidArgumentException;
+use JsonSerializable;
+use Psr\Http\Message\StreamInterface;
+
+/**
+ * A Datastore Entity
+ *
+ * Entity implements PHP's [ArrayAccess](http://php.net/arrayaccess), allowing
+ * access via the array syntax (example below).
+ *
+ * Example:
+ * ```
+ * use Google\Cloud\ServiceBuilder;
+ *
+ * $cloud = new ServiceBuilder;
+ * $datastore = $cloud->datastore();
+ *
+ * $key = $datastore->key('Person', 'Bob');
+ * $entity = $datastore->entity($key), [
+ *     'firstName' => 'Bob',
+ *     'lastName' => 'Testguy'
+ * ]);
+ *
+ * echo $entity['firstName']; // 'Bob'
+ * $entity['location'] = 'Detroit, MI';
+ * ```
+ */
+class Entity implements JsonSerializable, ArrayAccess
+{
+    use DatastoreTrait;
+
+    /**
+     * @var Key
+     */
+    private $key;
+
+    /**
+     * @var array
+     */
+    private $entity;
+
+    /**
+     * @var array
+     */
+    private $options;
+
+    /**
+     * @param Key $key The Entity's Key, defining its unique identifier
+     * @param array $entity The entity body
+     * @param array $options {
+     *     Configuration Options
+     *
+     *     @type bool $encode If set to true, some values will be base64_encoded
+     *     @type string $cursor Set only when the entity is obtained by a query
```

Comment:
What do you think about explicitly documenting `$encode` `$excludeFromIndexes` and `$className` here:

https://github.com/GoogleCloudPlatform/google-cloud-php/pull/101/files#diff-2124f3325a11419ecc0faf5cd6e8c598R275

instead of linking to this? It doesn't seem completely clear that `$options` on `DatastoreClient::entity()` relates to `$options` in the `Entity` constructor - especially if there are additional options documented here a user doesn't need to worry themselves with.